### PR TITLE
Actually setup gopath properly in go_genrule

### PR DESCRIPTION
**Pull request by mikedanese**
We require consumers to do a bunch of crazy stuff in go_genrule because we don't setup the gopath properly.

For example here:

https://github.com/kubernetes/kubernetes/blob/664edf832777cb7d6d00d38ccbcd4acba1497dc1/staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD#L41

Notice the `k8s.io/kubernetes/vendor` in `k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/sets/types`. That shouldn't be necessary. This fixes that by setting the gopath up properly.

### CODE REVIEW

File: defs/go.bzl [hunk 1]
```diff
@@ -12,87 +12,87 @@
...
-    for dep in ctx.attr.go_deps:
-        dep_import_paths.append(dep[GoLibrary].importpath)
+            args = ctx.actions.args()
+            args.add(srcfile.path)
+            args.add(target.path)
 
-    for go_path in ctx.attr.go_paths:
-        transitive_depsets.append(go_path.files)
-        label_dict[go_path.label] = go_path.files.to_list()
+            ctx.actions.run(
```
Comment (mikedanese) [inline defs/go.bzl]: This generates a lot of actions. It's not slow (e.g. takes a couple seconds to setup the gopath for the main openapi generation, which has a massive number of dependencies), but it cold be made faster.

**Comment by BenTheElder** [comment]
I _think_ this is right but I'm also not sure I'm confident enough in my grokking of bazel macro / rule implementations.
/assign @fejta

**Comment by mikedanese** [comment]
Ping.

**Comment by mikedanese** [comment]
Ping. Traceroute.

**Comment by fejta** [comment]
Mike are you confident this works?

**Comment by fejta** [comment]
Really need to convert this to a prow job

**Comment by mikedanese** [comment]
/hold

I'm going to drop the CROSSTOOL testing in travis.

**Comment by mikedanese** [comment]
@fejta to demonstrate that the crosstool stuff is broken, I've removed it in that last commit. The tests pass here and I've also tested this change in k/k. crosstools were dropped from bazel a few versions ago. We should probably remove support for cross compiling until someone has time to dedicate to it.

**Pull request merged**